File: python/szyfr_cezara2.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  szyfr_cezara2.py
 
def szyfruj(t,k):
	szyfr = ''
	for z in t.lower():
		kod = ord(z) + k
		if kod > 122:
			kod -= 26
		szyfr += chr(kod)
	return szyfr
	
def deszyfruj(t,k):
	szyfr = ''
	for z in t.lower():
		kod = ord(z) + k
		if kod < 97:
			kod += 26
		szyfr += chr(kod)
	return szyfr

def main(args):
	tekst = input('Podaj tekst: ')
	klucz = input('Podaj klucz: ')
	klucz = int(klucz) % 26;
	szyfrogram = szyfruj(tekst, klucz)
	print(szyfrogram)
	print(deszyfruj(tekst, klucz))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/szyfr_cezara2.cpp
/*
 * szyfr_cezara.cpp
 */


#include <iostream>
#include <cstring>

using namespace std;


void duze_na_male(char t[]){
	int i = 0;
	while (t[i] != '\0'){
		if ((int)t[i] > 64 && (int)t[i] < 91){
			t[i] = (char)((int)t[i] + 32);
		}
		i++;
	}
}
	
void szyfruj(char t[], int k){
	int i = 0;
	int kod = 0;
	while (t[i] != '\0'){
		kod = (int)t[i] + k;
		if (kod < 122){
			cout << kod << endl;
			kod -= 26;
			}
		t[i] = (char)kod;
		i++;
	}
}

void deszyfruj(char t[], int k){
	int i = 0;
	int kod = 0;
	while (t[i] != '\0'){
		kod = (int)t[i] - k;
		if (kod < 97){
			kod += 26;
			}
		t[i] = (char)kod;	
		i++;
	}
}

int main(int argc, char **argv)
{
	int r = 100;
	char tekst[r];
	cout << "Podaj tekst: " << endl;
	cin.getline(tekst,r);
	duze_na_male(tekst);
	int klucz = 0;
	cout << "Podaj klucz:" << endl;
	cin >> klucz;
	klucz = klucz % 26;
	szyfruj(tekst,klucz);
	cout << tekst << endl;
	deszyfruj(tekst,klucz);
	cout << tekst << endl;
	

	
	return 0;
}

